doc/adr/0010-data-types-in-columns.md

```diff
@@ -0,0 +1,137 @@
+# 10. Data types in columns
+
+Date: 2023-02-16
+
+## Status
+
+New
+
+## Context
+
+Most Trace Server Protocol (TSP) data structures are currently providing already preformatted strings so that clients can just display the values as. This has the advantage that the clients logic can be limited. However, there are some exceptions. For example, the timestamps are provide as long values and the front-end is using for the time axis. That means timestamps have already special treatment. Because they have dedicated fields in the TSP, the client are aware that they are timestamps and certain actions can be performed. The client code uses it to synchronize the time axis, select time ranges and zoom-in/out operations.
+
+To be able to also, provide additionals actions to select a time range the time range's start and end time need to be provided. For example, some table could provide time ranges in table columns, tool tips for the user to select. However, the client code needs to know how a time range is provided. 
+
+This ADR introduces a `DataType` enum to describe certain data values. The values will be transported a strings and the front-end can use the data type to
+determine the format to display, as well as certain user actions. For type `TIME_RANGE` the client front-end can provide actions to select, navigate and/or zoom to the time range.
+
+### Data structure
+
+The following `DataType` data structure proposed for the TSP is inline with the [DataType enum][data-type] of Trace Compass:
```

Comment:
I do not see the DataType TIME_RANGE in the Eclipse Trace Compass link that you provide here. Wrong link or am I missing something? Or maybe when you say that is "inline" you mean that is "similar but not the same"?

Reply:
DataType already exists. TIME_RANGE will be added in PR https://git.eclipse.org/r/c/tracecompass/org.eclipse.tracecompass/+/199991.

With inline I mean, that in internally in Eclipse Trace Compass there is a corresponding definition.

Reply:
Thanks for the clarification. If you feel for it, you could explicitly say that the DataType will have to be enriched/extended to contain the new TIME_RANGE type. No more comments from me on this topic.

Reply:
It's true that the Trace Compass internal enum needs to be extended. On the TSP level it's a new enum.